===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ argcv_string:n$34 == { 0 } }) free(*(argcv_string:buffer)) at 34 (line 175, column 16) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/inetutils-1.9.4/libinetutils/argcv.c
+++ b/table1/benchmarks/inetutils-1.9.4/libinetutils/argcv.c
@@ -172,7 +172,10 @@ argcv_string (int argc, char **argv, char **pstring)
   for (len = i = 0; i < argc; i++)
     {
       len += strlen (argv[i]) + 2;
-      buffer = realloc (buffer, len);
+      tmp = realloc(buffer, len)
+      if (tmp == NULL)
+        free(buffer);
+      buffer = tmp;
       if (buffer == NULL)
 	return 1;
       if (i != 0)
